src/b-splines/cubic.jl

```diff
@@ -0,0 +1,235 @@
+    m = length(args)
+    rowspec = spzeros(T, n, m)
+    colspec = spzeros(T, m, n)
+    valspec = zeros(T, m, m)
+
+    ix = 1
+    for (i, (row, col, val)) in enumerate(args)
+        rowspec[row, ix] = 1
+        colspec[ix, col] = 1
+        valspec[ix, ix] = val
+        ix += 1
+    end
+
+    rowspec, valspec, colspec
+end
+
+function define_indices_d{BC}(::Type{BSpline{Cubic{BC}}}, d, pad)
+    symix, symixm, symixp = symbol("ix_",d), symbol("ixm_",d), symbol("ixp_",d)
+    symixpp, symx, symfx = symbol("ixpp_",d), symbol("x_",d), symbol("fx_",d)
+    quote
+        # ensure that all three ix_d, ixm_d, and ixp_d are in-bounds no matter
+        # the value of pad
+        # TODO: Not sure if I'm handling the pad correctly, but relative to
+        #       quadratic case I did move upper clamp from `_ - 1` to `_ - 2`
+        #       to make room for the `pp` values
+        $symix = clamp(round(Int, real($symx)), 2-$pad, size(itp,$d)+$pad-2)
+        $symfx = $symx - $symix
+        $symix += $pad # padding for oob coefficient
+        $symixm = $symix - 1
+        $symixp = $symix + 1
+        $symixpp = $symixp + 1
+    end
+end
+
+# TODO: not sure if I am doing any of this Periodic stuff correctly
+function define_indices_d(::Type{BSpline{Cubic{Periodic}}}, d, pad)
+    symix, symixm, symixp = symbol("ix_",d), symbol("ixm_",d), symbol("ixp_",d)
+    symixpp, symx, symfx = symbol("ixpp_",d), symbol("x_",d), symbol("fx_",d)
+    quote
+        $symix = clamp(round(Int, real($symx)), 1, size(itp,$d))
+        $symfx = $symx - $symix
+        $symixm = mod1($symix - 1, size(itp,$d))
+        $symixp = mod1($symix + 1, size(itp,$d))
+        $symixpp = mod1($symix + 2, size(itp,$d))
+    end
+end
+
+# TODO: implement versions of `define_indices_d` for different Bc's if necessary
+
+padding{BC<:Flag}(::Type{BSpline{Cubic{BC}}}) = Val{1}()
+padding(::Type{BSpline{Cubic{Periodic}}}) = Val{0}()
+
+"""
+In this function we assume that `fx_d = x-ix_d` and we produce `cX_d` for
+`X ⋹ {m, _, p, pp}` such that
+
+    cm_d  = p(fx_d)
+    c_d   = q(fx_d)
+    cp_d  = q(1-fx_d)
+    cpp_d = p(1-fx_d)
+
+where `p` and `q` are defined in the docstring entry for `Cubic`.
+"""
+function coefficients{C<:Cubic}(::Type{BSpline{C}}, N, d)
+    symm, sym =  symbol("cm_",d), symbol("c_",d)
+    symp, sympp = symbol("cp_",d) ,symbol("cpp_",d)
+    symfx = symbol("fx_",d)
+    symfx_cub = symbol("fx_cub_", d)
+    sym_1m_fx_cub = symbol("one_m_fx_cub_", d)
+    quote
+        $symfx_cub = cub($symfx)
+        $sym_1m_fx_cub = cub(1-$symfx)
+        $symm = SimpleRatio(1,6)*$sym_1m_fx_cub
+        $sym  = SimpleRatio(2,3) - sqr($symfx) + SimpleRatio(1,2)*$symfx_cub
+        $symp = SimpleRatio(2,3) - sqr(1-$symfx) + SimpleRatio(1,2)*$sym_1m_fx_cub
+        $sympp = SimpleRatio(1,6)*$symfx_cub
+    end
+end
+
+# This assumes integral values ixm_d, ix_d, and ixp_d,
+# coefficients cm_d, c_d, and cp_d, and an array itp.coefs
```

Comment:
This comment seems like it's copied from the comment above the corresponding quadratic method. Should probably be either updated (and converted to a docstring?) or removed entirely.